csi/node.go

```diff
@@ -246,16 +246,101 @@ func (s *OsdCsiServer) NodeGetCapabilities(
 
 	logrus.Debugf("csi.NodeGetCapabilities request received")
 
-	return &csi.NodeGetCapabilitiesResponse{
-		Capabilities: []*csi.NodeServiceCapability{
-			{
-				Type: &csi.NodeServiceCapability_Rpc{
-					Rpc: &csi.NodeServiceCapability_RPC{
-						Type: csi.NodeServiceCapability_RPC_UNKNOWN,
-					},
+	caps := []csi.NodeServiceCapability_RPC_Type{
+		// Getting volume stats for volume health monitoring
+		csi.NodeServiceCapability_RPC_GET_VOLUME_STATS,
+		// Indicates that the Node service can report volume conditions.
+		csi.NodeServiceCapability_RPC_VOLUME_CONDITION,
+	}
+
+	var serviceCapabilities []*csi.NodeServiceCapability
+	for _, cap := range caps {
+		serviceCapabilities = append(serviceCapabilities, &csi.NodeServiceCapability{
+			Type: &csi.NodeServiceCapability_Rpc{
+				Rpc: &csi.NodeServiceCapability_RPC{
+					Type: cap,
 				},
 			},
+		})
+	}
+
+	return &csi.NodeGetCapabilitiesResponse{
+		Capabilities: serviceCapabilities,
+	}, nil
+}
+
+func getVolumeCondition(vol *api.Volume) *csi.VolumeCondition {
+	condition := &csi.VolumeCondition{}
+	if vol.Status != api.VolumeStatus_VOLUME_STATUS_UP {
+		condition.Abnormal = true
+	}
+
+	switch vol.Status {
+	case api.VolumeStatus_VOLUME_STATUS_UP:
+		condition.Message = "Volume status is up"
+
+	case api.VolumeStatus_VOLUME_STATUS_NOT_PRESENT:
+		condition.Message = "Volume status is not present"
+
+	case api.VolumeStatus_VOLUME_STATUS_DOWN:
+		condition.Message = "Volume status is down"
+
+	case api.VolumeStatus_VOLUME_STATUS_DEGRADED:
+		condition.Message = "Volume status is degraded"
+
+	default:
+		condition.Message = "Volume status is unknown"
+	}
+
+	return condition
+}
+
+// NodeGetVolumeStats get volume stats for a given node
```

Comment:
Add here information about security just like the others

Reply:
Added